tapdb/asset_minting.go

```diff
@@ -964,27 +964,67 @@ func fetchAssetMetas(ctx context.Context, db PendingAssetStore,
 	for _, assetT := range assets {
 		assetID := assetT.ID()
 
-		assetMeta, err := db.FetchAssetMetaForAsset(ctx, assetID[:])
-		switch {
+		assetMeta, err := fetchMetaByAssetID(ctx, db, assetID)
+		if err != nil {
+			return nil, err
+		}
+
 		// If the asset doesn't have a meta data reveal, then we can
 		// skip it.
-		case errors.Is(err, sql.ErrNoRows):
+		if assetMeta == nil {
 			continue
-
-		case err != nil:
-			return nil, err
 		}
 
 		scriptKey := asset.ToSerialized(assetT.ScriptKey.PubKey)
-		assetMetas[scriptKey] = &proof.MetaReveal{
-			Data: assetMeta.MetaDataBlob,
-			Type: proof.MetaType(assetMeta.MetaDataType.Int16),
-		}
+		assetMetas[scriptKey] = assetMeta
 	}
 
 	return assetMetas, nil
 }
 
+// fetchMetaByAssetID fetches a single asset meta reveal.
+func fetchMetaByAssetID(ctx context.Context, db PendingAssetStore,
+	ID asset.ID) (*proof.MetaReveal, error) {
+
+	assetMeta, err := db.FetchAssetMetaForAsset(ctx, ID[:])
+	switch {
+	case errors.Is(err, sql.ErrNoRows):
+		return nil, nil
```

Comment:
I think this sort of case is where we want to `return fn.None[proof.MetaReveal](), nil`
But what you have should work so it might do for now. Maybe add a TODO?